### DISCUSSION TITLE
Time correction in incoming data

### DISCUSSION THREAD

**Discussion by Zwioor** [macxq/foxess-ha#245] [General]
I have a problem with a 1-hour time shift in incoming data. Does anyone know what causes this error and is it possible to correct the time in the data? This causes further problems in the energy section. The energy generated at 7:00 in the fox panel is displayed at 6:00 in the HA panel, etc. I'm sorry if I missed the solution to such a problem.

![fox](https://github.com/macxq/foxess-ha/assets/155753961/e53b392b-d3dc-4a86-8ce7-34acc965baba)
![HA](https://github.com/macxq/foxess-ha/assets/155753961/ac355543-6c00-49e5-b4e9-54ab03dd5de5)

**Comment by FozzieUK** [comment]
I'm not sure there is a time shift?, the statistics graph from the Fox website displays badly because the time label is in the wrong place. For example the graph starts from `1` but what it means is the first hour i.e. `00:00 to 00:59`, and it then increases an hour at a time - so where the label says `15` it is actually `14:00 to 14:59` and the element that says `24` is actually `23:00 to 23:59`.

I've modified the graph axis to shift the labels across the way it should display and that looks correct to me now ?
![325239227-e53b392b-d3dc-4a86-8ce7-34acc965baba](https://github.com/macxq/foxess-ha/assets/113460294/c94977df-cdc8-4c10-818f-7551f30d86d8)

**Reply by Zwioor** [reply]
I didn't explain clearly, the time shift applies to the data sent by the tauronamiplus plugin, which makes the data inconsistent, or at least that's what I think. For example, when at 12:00 Taurons Plus sends 3.8, Fox sends 2.6 but at 11:00 Fox sends 3.8.
![ha2](https://github.com/macxq/foxess-ha/assets/155753961/c9589898-066c-4493-97b0-c792b421819a)

**Reply by Zwioor** [reply]
Actually, looking at these charts, I'm not sure what's causing the data to be inconsistent.

**Reply by FozzieUK** [reply]
I'm not actually sure what is causing this, the fox cloud data is correct as the data it stores it in UTC and it converts it to your local timezone for display.

This foxcloud integration doesn't gather any time information from the fox cloud, it gathers the data in near real time. When the entities are updated it will come from the very latest device variable sample (sent by your datalogger) and it will only be timestamped at that point with the timebase of the home asssistant you are running on.

**Reply by Zwioor** [reply]
It is possible that the data sent by tauronamiplus is shifted. I'll try to check it out.
